src/Development/IDE/Core/Compile.hs

```diff
@@ -144,9 +145,14 @@ compileModule packageState deps tmr =
                 let pm' = pm{pm_mod_summary = tweak $ pm_mod_summary pm}
                 let tm' = tm{tm_parsed_module  = pm'}
                 GHC.dm_core_module <$> GHC.desugarModule tm'
-
+            let tc_result = fst (tm_internals_ (tmrModule tmr))
```

Comment:
Do we need to set `-O0` explicitly or does `hie-bios` set that?

Reply:
I was under the impression that `ghcide` set all the flags it was interested in. At least on the `.hi` files branch, I believe this is the case.

Reply:
I don’t know any code that sets `-O0`. I think so far it just didn’t matter since we didn’t run the simplifier and don’t do code generation.

Reply:
`hie-bios` definitely sets it: https://github.com/mpickering/hie-bios/blob/master/src/HIE/Bios/Environment.hs#L41
Assuming, `ghcide` still uses `initSession`.

Reply:
On the .hi file branch this will need to be set but for now, `initSession` does set it.